core/domain/feedback_services.py

```diff
@@ -261,3 +262,27 @@ def enqueue_feedback_message_email_task(user_id):
     taskqueue_services.enqueue_task(
         feconf.FEEDBACK_MESSAGE_EMAIL_HANDLER_URL, {'user_id': user_id},
         feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_COUNTDOWN_SECS)
+
+
+def send_feedback_message_email(exploration_id, thread_id, message_id):
+    """Creates"""
+    exploration_rights = rights_manager.get_exploration_rights(exploration_id)
+    feedback_message = {
+        'exploration_id': exploration_id,
+        'thread_id': thread_id,
+        'message_id': message_id
+    }
+
+    for owner in exploration_rights.owner_ids:
+        model = feedback_models.UnsentFeedbackEmailModel.get(
+            owner, strict=False)
+
+        if model:
+            model.feedback_messages.append(feedback_message)
+            model.put()
+
+        else:
+            model = feedback_models.UnsentFeedbackEmailModel(
```

Comment:
Ok, how about this solution:
Create a separate function for body part of for loop, we will call it through transaction services.
So this solves first issue: if Thread 1 gets the model. Thread 2 cannot possibly get it and write into it. Only when Thread 1 is done with writing, Thread 2 will be allowed to modify the model.

For 2nd issue: we can make a separate function, which will get model and delete it in same function and return the information stored in model. We can make call to this function usinf transaction services. So now if Thread 2 gets the model, Thread 1 will have to wait until model is deleted by Thread 2 and then Thread 1 will see that model is deleted and so it will create a new model. And if Thread 1 tries to modify (as your scenario states) then Thread 2 will have to wait until it can get or delete model.

I think this solution can work without race conditions.
Only one thing is confusing at this point in this solution. I talked about 2 different functions, first get_and_create and other is get_and_delete, what confusing is that, can these 2 functions run in transaction or we need a single function ? 
Because we can allow only one function to be executed and only by one thread at a time. Function 1 should not execute when Thread 2 is executing function 2 and vice versa.